runelite-client/src/main/java/net/runelite/client/plugins/virtuallevels/VirtualLevelsPlugin.java

```diff
@@ -27,45 +27,106 @@
 
 import com.google.common.eventbus.Subscribe;
 import javax.inject.Inject;
+
+import com.google.inject.Provides;
 import net.runelite.api.Client;
 import net.runelite.api.Experience;
 import net.runelite.api.Skill;
+import net.runelite.api.WorldType;
+import net.runelite.api.events.ConfigChanged;
 import net.runelite.api.events.ScriptCallbackEvent;
+import net.runelite.api.events.WidgetHiddenChanged;
+import net.runelite.api.widgets.Widget;
+import net.runelite.api.widgets.WidgetInfo;
 import net.runelite.client.callback.ClientThread;
+import net.runelite.client.config.ConfigManager;
 import net.runelite.client.events.PluginChanged;
 import net.runelite.client.plugins.Plugin;
 import net.runelite.client.plugins.PluginDescriptor;
 
+import java.util.ArrayList;
+
 @PluginDescriptor(
 	name = "Virtual Levels",
 	description = "Configuration for the virtual levels plugin.",
-	tags = {"skill", "total", "max"},
+	tags = {"skill", "total", "max", "f2p", "free "},
 	enabledByDefault = false
 )
+
 public class VirtualLevelsPlugin extends Plugin
 {
 	private static final String TOTAL_LEVEL_TEXT_PREFIX = "Total level:<br>";
 
+	private static final int
```

Comment:
As mentioned in Discord, these positions should be taken from the widgets that they're taking the place of, not hard-coded.

Reply:
These coordinates aren't the widgets they are taking place of, they are the normal location of the widgets. But if I use getRelativeX() for these widgets, the value will be incorrect if the plugin is enabled before logging in. It's also tricky to get these values as the stat widgets don't load on game start.

Reply:
It seems like you could set the "known positions" to `-1` until you actually find the locations, and try to find them on `GameStateChanged` or `GameTick` events until they're found.